enhancements/network/ip-interface-selection.yaml

```diff
@@ -0,0 +1,302 @@
+---
+title: ip-interface-selection
+authors:
+  - @cybertron
+reviewers:
+  - @jcaamano
+  - @tsorya
+approvers:
+  - TBD
+api-approvers:
+  - "None"
+creation-date: 2022-07-07
+last-updated: 2022-07-07
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - https://github.com/openshift/baremetal-runtimecfg/issues/119
+replaces:
+superseded-by:
+---
+
+# IP and Interface Selection
+
+## Summary
+
+As OpenShift is deployed in increasingly complex networking environments, we
+have gotten many requests for more control over which interface is used for
+the primary node IP. We provided a basic mechanism for this with
+[KUBELET_NODEIP_HINT](https://github.com/openshift/machine-config-operator/pull/2888)
+but as users have started to exercise that some significant limitations have
+come to light.
+
+## Motivation
+
+Some users want to have a great deal of control over how their network traffic
+is routed. Because we use the default route for interface and IP selection,
+in some cases they are not able to route traffic the way they want.
+
+### User Stories
+
+As a network administrator, I want cluster traffic to be on one NIC and the
+default route on a different NIC so external traffic is segregated from
```

Comment:
`s/default route/default route(s)/`  There can be more that one default route, per stack, or also multi-homed with a metric.

Reply:
Right, this should probably say _primary_ default route, because for the purposes of IP selection we only care about the highest priority default route. We can only select one IP/interface so the others are ignored.

I'm not at all certain how that additionalNetworkIDs feature works today. We don't support deployment across multiple subnets so all of your masters have to be in the same network. That's part of why I'm focused on UPI in this document. We don't support overriding the IP selection logic for IPI anyway, it's always based on the VIP.